### PR TITLE
v4.1.x: mtl/ofi: Use fi_tinject if possible in isend

### PR DESCRIPTION
**Pull request by mjkoop**
Attempt to use fi_tinject for messages within
isend to reduce the overhead when a message can
be immediately buffered and sent.

### CODE REVIEW

File: ompi/mca/mtl/ofi/mtl_ofi.h [hunk 1]
```diff
@@ -612,6 +612,44 @@ ompi_mtl_ofi_isend_generic(struct mca_mtl_base_module_t *mtl,
...
+                                    start,
+                                    length,
+                                    sep_peer_fiaddr,
+                                    match_bits);
+        }
+
+        if(OPAL_LIKELY(ret == 0)) {
+            ofi_req->event_callback(NULL, ofi_req);
+            return ofi_req->status.MPI_ERROR;
+        } else if(ret != -FI_EAGAIN) {
```
Comment (wzamazon) [inline ompi/mca/mtl/ofi/mtl_ofi.h:638]: `else` seems to be unnecessary

**Review by wzamazon — APPROVED** [review]

**Comment by bwbarrett** [comment]
Why does this help anything?  We have to create the request anyway and the user has to call test anyway, so it seems like any real help is due to an issue in the Libfabric provider, or am I missing something?

**Comment by mjkoop** [comment]
> Why does this help anything? We have to create the request anyway and the user has to call test anyway, so it seems like any real help is due to an issue in the Libfabric provider, or am I missing something?

It prevents needing to add the CQE, poll, and pull the CQE out, which is a small but measurable impact.

**Comment by bwbarrett** [comment]
Putting in draft until https://github.com/open-mpi/ompi/pull/11584 is done (since this should be a backport).

**Comment by hppritcha** [comment]
what is the corresponding commit on main?  I'm not interested in testing this PR on HPE CXI as we don't support the target release (v4.1.x) with the CXI provider.

**Comment by bwbarrett** [comment]
I'm going to close this, since https://github.com/open-mpi/ompi/pull/11584 is the same change on main and we should follow the usual flow.  We'll backport when it is committed to main.

**Pull request closed without merging**